### PR TITLE
[SPARK-43582][BUILD] Upgrade `sbt-pom-reader` to 2.4.0

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aims to upgrade `sbt-pom-reader` from 2.2.0 to 2.4.0.

### Why are the changes needed?
Since v2.3.0, organization has moved from `com.typesafe.sbt` to `com.github.sbt`
- https://github.com/sbt/sbt-pom-reader/releases/tag/v2.4.0
- https://github.com/sbt/sbt-pom-reader/releases/tag/v2.3.0

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (Pending CI).

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Oh, could you take a look at the build error? This looks like a breaking change.
```
[error] /home/runner/work/spark/spark/project/SparkBuild.scala:34:21: object sbt is not a member of package com.typesafe
[error] import com.typesafe.sbt.pom.{PomBuild, SbtPomKeys}
```

**Comment by panbingkun** [comment]
> Oh, could you take a look at the build error? This looks like a breaking change.
> 
> ```
> [error] /home/runner/work/spark/spark/project/SparkBuild.scala:34:21: object sbt is not a member of package com.typesafe
> [error] import com.typesafe.sbt.pom.{PomBuild, SbtPomKeys}
> ```

Ok, let me check it.

**Comment by dongjoon-hyun** [comment]
Thank you for update!

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -31,7 +31,7 @@ import sbt.Keys._
 import sbt.librarymanagement.{ VersionNumber, SemanticSelector }
 import com.etsy.sbt.checkstyle.CheckstylePlugin.autoImport._
 import com.simplytyped.Antlr4Plugin._
-import com.typesafe.sbt.pom.{PomBuild, SbtPomKeys}
+import sbtpomreader.{PomBuild, SbtPomKeys}
```
Comment (panbingkun) [inline project/SparkBuild.scala:34]: https://github.com/sbt/sbt-pom-reader/blob/main/src/main/scala/sbtpomreader/PomBuild.scala#L1~L9

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**